### ISSUE TITLE
The "room-opened" event is not fired in iframe under special conditions

### ISSUE THREAD

**Issue by mahtifarahani** [RocketChat/Rocket.Chat#31327]
Description:
Steps to reproduce:

In the Windows Chrome browser in Iframe Rocket Chat, if you open a chat room and go to another chat room, after returning to the previous chat, you will see that this event is not sent ("room-opened"), because maybe that page has already been loaded, if not created Please switch between the two chats several times. You can see that this event is sometimes not fired

Expected behavior:
This event is fired every time you navigate between two chat room

Server Setup Information:
Version of [Rocket.Chat](https://rocket.chat/) Server: 6.5.0
Operating System: ubuntu 22.04
Deployment Method: tar
Number of Running Instances: 1
NodeJS Version: 14.21.3
MongoDB Version: 5

Client Setup Information
Chrome Version 120.0.6099.71 (Official Build) (64-bit)
Operating System: windows 11

**Comment by Sayan4444** [comment]
@mahtifarahani Can you please share the video the issue? It will be of greate help

**Comment by Sayan4444** [comment]
https://github.com/RocketChat/Rocket.Chat/assets/112304873/bbf270a4-2d81-449d-8779-97b2ab0fa5ca
I just tried to replicate your issue several time but failed. Can you tell me the specific User Experience issue that the end user will face even if the 'event' is missed under special conditions. Awating for your video

**Comment by Devansht420** [comment]
@mahtifarahani maybe try to use some other browser and also share a video of the particular problem
as i am also unable to replicate the issue like @Sayan4444

**Comment by mahtifarahani** [comment]
@Sayan4444 @Devansht420 
Thank you for your attention
Uploading rc2.mp4…

**Comment by gabriellsh** [comment]
Hi! First of all, thanks for contributing!

I don't know if we sould categorize this as a bug, because the reason the event only fires once is that we cache up to 5 rooms, so if you navigate between those it will only dispatch the event only the first time per room.

You can test this by opening 5 rooms in sequence and navigating in between them. If you open one more, the least-recently openend room will be uncached and the event will dispatch again once you open it.

Do you know by any chance if this behaviour was different in another version?

**Comment by mahtifarahani** [comment]
> Hi! First of all, thanks for contributing!
> 
> I don't know if we sould categorize this as a bug, because the reason the event only fires once is that we cache up to 5 rooms, so if you navigate between those it will only dispatch the event only the first time per room.
> 
> You can test this by opening 5 rooms in sequence and navigating in between them. If you open one more, the least-recently openend room will be uncached and the event will dispatch again once you open it.
> 
> Do you know by any chance if this behaviour was different in another version?

Yes, it was like this in previous versions.  We used to use this event to get information about the new page, but now we don't have any information about the new page.  In addition, this event sometimes remains and returns wrong and earlier values

**Comment by mahtifarahani** [comment]
> @mahtifarahani does my PR solves the problem?

Yes, this problem is solved, thank you

**Comment by Sayan4444** [comment]
@mahtifarahani  Great, but its not merged yet, lets see what happens
